src/HotChocolate/Fusion-vnext/src/Fusion.Composition/PreMergeValidation/Rules/ExternalArgumentDefaultMismatchRule.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections.Immutable;
+using HotChocolate.Fusion.Events;
+using static HotChocolate.Fusion.Logging.LogEntryHelper;
+
+namespace HotChocolate.Fusion.PreMergeValidation.Rules;
+
+/// <summary>
+/// This rule ensures that arguments on fields marked as <c>@external</c> have default values
+/// compatible with the corresponding arguments on fields from other source schemas where the field
+/// is defined (non-<c>@external</c>).
+/// </summary>
+/// <seealso href="https://graphql.github.io/composite-schemas-spec/draft/#sec-External-Argument-Default-Mismatch">
+/// Specification
+/// </seealso>
+internal sealed class ExternalArgumentDefaultMismatchRule : IEventHandler<OutputFieldGroupEvent>
+{
+    public void Handle(OutputFieldGroupEvent @event, CompositionContext context)
+    {
+        var (fieldName, fieldGroup, typeName) = @event;
+
+        var externalFields = fieldGroup
+            .Where(i => ValidationHelper.IsExternal(i.Field))
+            .ToImmutableArray();
+
+        if (externalFields.Length == 0)
+        {
+            return;
+        }
+
+        var argumentNames = fieldGroup
+            .SelectMany(i => i.Field.Arguments, (_, arg) => arg.Name)
+            .ToHashSet();
+
+        foreach (var argumentName in argumentNames)
+        {
+            var arguments = fieldGroup
+                .Select(i => i.Field.Arguments[argumentName])
```

Comment:
Won't this fail if the field doesn't have this argument?

Reply:
Yes, this would fail. We could add a `GetOrDefault()` extension method if that's needed or do an inner loop